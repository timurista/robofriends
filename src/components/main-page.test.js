import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './main-page'

const initialProps = {
  searchField: '',
  robots: [],
  isPending: false,
  error: false,
  onSearchChange: jest.fn(Promise.resolve()),
  onRequestRobots: jest.fn().mockReturnValue(Promise.resolve())
}

const setup = (props: {}) => {
  const newProps = {...initialProps, ...props}
  return shallow(<MainPage {...newProps} />)
}

it('renders the app', () => {
  const wrapper = setup()
  expect(wrapper.debug()).toMatchSnapshot()
})

it('filters robots and returns empty array', () => {
  const wrapper = setup()
  expect(wrapper.instance().filteredRobots([])).toEqual([])
})

it('filters robots and returns empty array', () => {
  const robots = [
    {id: 3, name: 'john', email: 'john@gmail.com'}
  ];
  const wrapper = setup({ searchField: 'a', robots })
  expect(wrapper
    .instance()
    .filteredRobots()
  ).toEqual([])
})

it('shows error', () => {
  const wrapper = setup({ error: 'error'});
  expect(wrapper.find('#error')).toHaveLength(1)
})

it('shows pending', () => {
  const wrapper = setup({ isPending: true});
  expect(wrapper.find('#loading')).toHaveLength(1)
})

it('filters robots and returns john only', () => {
  const robots = [
    {id: 3, name: 'john', email: 'john@gmail.com'},
    {id: 3, name: 'ggg', email: 'ggg@gmail.com'},
  ];
  const wrapper = setup({ searchField: 'j', robots })
  expect(wrapper
    .instance()
    .filteredRobots()
  ).toEqual([{"email": "john@gmail.com", "id": 3, "name": "john"}])
})