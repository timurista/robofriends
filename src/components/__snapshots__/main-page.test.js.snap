// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox onSearchChange={[Function: mockConstructor]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robots={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
